src/fetch_metrics.py

```diff
@@ -0,0 +1,250 @@
+import os
+import requests
+import yaml
+import warnings
+import string
+import argparse
+from collections import defaultdict
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from urllib3.exceptions import InsecureRequestWarning
+
+# Suppress SSL warnings
+warnings.simplefilter('ignore', InsecureRequestWarning)
+
+# Prometheus server URL
+prometheus_url = "https://prometheus-k8s-openshift-monitoring.apps.cnv2.engineering.redhat.com"
+
+# Path to CA certificate (optional)
+ca_cert_path = None  # Replace with "./tls.crt" if you have a CA certificate
+
+# Directory for saving output files
+output_dir = "docs/prometheus_metrics"
+
+# Path to label descriptions file
+label_descriptions_file = "data/label_descriptions.yaml"
+
+def parse_arguments():
+    """
+    Parse command-line arguments.
+    """
+    parser = argparse.ArgumentParser(description="Fetch and update OpenShift Prometheus metrics documentation.")
+    parser.add_argument("--token", required=True, help="Authentication token for accessing the Prometheus API.")
+    return parser.parse_args()
+
+def load_label_descriptions(file_path):
+    """
+    Load label descriptions from a YAML file.
+    """
+    try:
+        with open(file_path, "r") as yamlfile:
+            labels = yaml.safe_load(yamlfile)
+            return {item['label_name']: item['label_description'] for item in labels}
+    except IOError as e:
+        print(f"Error reading label descriptions file: {e}")
+        return {}
+
+
+# Retry logic for HTTP requests
+def make_request_with_retries(url, params=None, headers=None, verify=True, retries=3, timeout=30):
```

Comment:
Might be worth using https://github.com/4n4nd/prometheus-api-client-python? It misses some endpoints, but provides easy boilerplating for things around auth, so we use it for ACM notebooks/scripts.

Reply:
We had similar needs in https://github.com/obsint/obsinthe/blob/main/obsinthe/prometheus/client.py, but there was just a very small amount of actual code we needed (+ we also needed different behavior in some cases), to eventually we ended up with this small client implementation instead. Frankly the prometheus API is quite straightforward. Also, I'm not sure about the maintanence state of prometheus-api-client-python

Reply:
Ack, makes sense, I see now it isn't super well maintained too!